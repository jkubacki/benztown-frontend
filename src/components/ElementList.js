import React from 'react';
import PropTypes from 'prop-types';
import getElementsAction from 'actions/elements';
import { connect } from 'react-redux';

import Element from 'components/Element';

class ElementList extends React.Component {
  componentWillMount() {
    const { getElements } = this.props;
    getElements({});
  }

  render() {
    const { elements } = this.props;
    return (
      <table className="table">
        <thead>
          <tr>
            <th scope="col">Name</th>
            <th scope="col">File</th>
            <th scope="col">Tags</th>
          </tr>
        </thead>
        <tbody>
          {elements.map(element => <Element key={element.id} element={element} />)}
        </tbody>
      </table>
    );
  }
}

ElementList.propTypes = {
  elements: PropTypes.arrayOf(
    PropTypes.shape({
      attributes: PropTypes.shape({
        name: PropTypes.string.isRequired,
        file: PropTypes.string.isRequired,
        tag_list: PropTypes.array.isRequired,
      }).isRequired,
    }).isRequired,
  ).isRequired,
  getElements: PropTypes.func.isRequired,
};

function mapStateToProps(props) {
  const { elements } = props;
  return { elements };
}

export { ElementList }

export default connect(
  mapStateToProps,
  { getElements: getElementsAction },
)(ElementList);
