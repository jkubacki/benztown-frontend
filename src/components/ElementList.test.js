import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ElementList } from './ElementList';

const props = {
  getElements: jest.fn(),
  elements: [
    { id: 1, attributes: { name: 'Element name1', file: 'file.mp3', tag_list: ['tag1', 'tag2', 'tag3'] } },
    { id: 2, attributes: { name: 'Element name2', file: 'file.mp3', tag_list: ['tag2', 'tag3', 'tag4'] } },
  ],
};

it('renders without crashing', () => {
  const component = shallow(<ElementList {...props} isLoggedIn />);
  expect(toJson(component)).toMatchSnapshot();
});
