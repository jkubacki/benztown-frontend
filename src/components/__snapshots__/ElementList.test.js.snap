// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Name
      </th>
      <th
        scope="col"
      >
        File
      </th>
      <th
        scope="col"
      >
        Tags
      </th>
    </tr>
  </thead>
  <tbody>
    <Element
      element={
        Object {
          "attributes": Object {
            "file": "file.mp3",
            "name": "Element name1",
            "tag_list": Array [
              "tag1",
              "tag2",
              "tag3",
            ],
          },
          "id": 1,
        }
      }
      key="1"
    />
    <Element
      element={
        Object {
          "attributes": Object {
            "file": "file.mp3",
            "name": "Element name2",
            "tag_list": Array [
              "tag2",
              "tag3",
              "tag4",
            ],
          },
          "id": 2,
        }
      }
      key="2"
    />
  </tbody>
</table>
`;
